### PR TITLE
RetireJS, node 8

### PR DESCRIPTION
**Pull request by SethDavenport**
Toolchain updates

* use node 8 as the build environment.
* add retireJS check to circleCI. RetireJS checks your dependencies and flags any with known security vulnerabilities.

Unfortunately retireJS flagged angularJS and jquery, prompting a major version update for each.

Updating Jquery broke autonumeric, which is used by ws-currency. I had to fix this by updating autonumeric to it's latest major version, which has a totally different API...

I also updated the peerDependencies to match; technically this changes the library's "interface" on npm so I made a major version bump.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,6 +1,3 @@
 machine:
   node:
-    version: 6.1.0
-dependencies:
```
Comment (SethDavenport) [inline circle.yml]: This is redundant. CircleCI infers it.

File: index.html [hunk 2]
```diff
@@ -9,10 +9,10 @@
          app will need to install too. I'm using CDNs here but for production you'll want to pull
          them in with npm or bower as appropriate -->
     <link rel="stylesheet" href="https://unpkg.com/@wealthsimple/fabric@3.1.0">
-    <script src="https://unpkg.com/jquery@2.1.4/dist/jquery.js"></script>
-    <script src="https://unpkg.com/angular@1.5.5/angular.js"></script>
-    <script src="https://unpkg.com/angular-ui-router@0.2.15/release/angular-ui-router.js"></script>
-    <script src="https://unpkg.com/angular-animate@1.5.5/angular-animate.js"></script>
+    <script src="https://unpkg.com/jquery@3.2.1/dist/jquery.js"></script>
+    <script src="https://unpkg.com/angular@1.6.5/angular.js"></script>
+    <script src="https://unpkg.com/@uirouter/angularjs@1.0.3/release/angular-ui-router.js"></script>
```
Comment (SethDavenport) [inline index.html]: the angular-ui-router folks renamed their package. new look; same great library

**Comment by benjyblack** [comment]
Angular 1.6 has some breaking changes, did we check to make sure they don't affect these components?

**Comment by SethDavenport** [comment]
Yes - I did an audit based on https://docs.angularjs.org/guide/migration#migrating-from-1-5-to-1-6

The only major flag for ws-directives specifically was the `$onInit` stuff - however it only affects components/directives that
* use isolate scopes or bindings
* use controllers
* access scope or binding stuff prior to `$onInit`

Turns out only two components fall into that category, `ws-expander` and `ws-signature-pad`; they are both using `$ngOnInit` already.

The other directives all seem to put their logic in the `link` function, with IIRC runs after `$compile`

Finally, I made sure that the unit tests work and the test page served by `npm run start:dev` functions without any errors.

It's unfortunate that 1.5.8 has security issues but 1.6 has so many breaking changes...

**Pull request merged**